server/game/Card.js

```diff
@@ -101,6 +102,17 @@ class Card extends EffectSource {
         this.endRound();
     }
 
+    get type() {
+        if(this.effectiveType !== null) {
+            return this.effectiveType;
+        }
+        return this.printedType;
+    }
+
+    set type(t) {
```

Comment:
type should never be set

Reply:
A card is a gameobject and all gameobjects set their type to '' in the constructor. As of this PR the only objects where it makes sense to mask the type is cards, but it won't compile if I don't let the constructor set the type. Should I move the masking into GameObject anyway - or is there a way I can get around the GameObject constructor trying to set the type?

Reply:
I dunno, this still feels like a smell to me - let me look at this in more detail to see if there's a better way of doing this

Reply:
+1

Reply:
Ok having looked at this further, this.type is only set to '' to keep the "typescript" autocomplete happy, so it seems perverse to introduce a setter just for that.  So let's remove the initialisation and the setter :)

Reply:
Done!